docker-compose.dev.yml

```diff
@@ -1,27 +1,82 @@
 services:
+  nginx:
+    image: nginx:latest
+    ports:
+      - 8080:8080
+    configs:
+      - source: nginx_config
+        target: /etc/nginx/nginx.conf
+    networks:
+      - public
+    depends_on:
+      server:
+        condition: service_healthy
+        restart: true
+
   server:
     build:
       context: ./server
       dockerfile: Dockerfile.dev
-    ports:
-      - '5001:5001'
+    # ports:
+    #   - 5001:5001
     volumes:
       - ./server:/app
       - /app/node_modules
     environment:
       - NODE_ENV=development
     env_file:
       - .env
+    networks:
+      - private
+      - public
+    depends_on:
+      db:
+        condition: service_healthy
+        restart: true
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://server:5001"]
+      interval: 5s
+      retries: 5
+      start_period: 5s
+      timeout: 5s
 
   db:
     image: postgres:latest
-    ports:
-      - '5432:5432'
-    env_file:
-      - .env
+    # ports:
+    #   - "5432:5432"
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_DB: acm_website
+      POSTGRES_PASSWORD_FILE: /run/secrets/pg_password
     volumes:
       - acm_website_db_data:/var/lib/postgresql/data
       - ./db-scripts:/docker-entrypoint-initdb.d/
+    secrets:
+      - pg_password
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
+      interval: 5s
+      timeout: 5s
+      retries: 5
+      start_period: 15s
+    networks:
+      - private
 
 volumes:
   acm_website_db_data:
+
+networks:
```

Comment:
Q: why would we need an external network for the purpose of local development

Reply:
isn't it better to simulate production environment

Reply:
we don't have to simulate production environment for the development `docker compose` because that will be more resource-consuming on the developer's machine.
The goal of this `docker compose ` is for the developers to work on code and to be as light-weighted as possible